handlers/stock.js

```diff
@@ -0,0 +1,176 @@
+
+                            updateUser(tableSvc, message.author.username, newSenderTotal);
+                            updateUser(tableSvc, recipient, newReceiverTotal);
+
+                            transactionString += `${message.author.username}: ${senderStock} - ${sendAmount} = ${newSenderTotal}\n
+                                                  ${recipient}: ${receiverStock} + ${sendAmount} = ${newReceiverTotal}`
+
+                            LOG.info(transactionString);
+                            channel.send(transactionString);
+                        }
+                        else {
+                            channel.send('Not enough stock to make the transaction.')
+                        }
+                    })
+                }
+                else {
+                    channel.send('That user does not exist! Schreef won\'t throw his stock to the abyss.');
+                }
+            })
+        }
+        else {
+            channel.send('You can\'t send yourself stock here.');
+        }
+    }
+
+};
+
+// Used to update or create a user entry in the Azure Table
+function updateUser(tableSvc, username, amount) {
+
+    let userEntry = {
+        PartitionKey: ENTGEN.String(username),
+        RowKey: ENTGEN.String('1'),
+        schreefstock: ENTGEN.Int32(amount),
+    }
+
+    tableSvc.insertOrReplaceEntity(TABLENAME, userEntry, null, function (error, result, response) {
+        if (!error) {
+            // Entity updated
+        }
+        else {
+            LOG.error(error);
+        }
+    })
+};
+
+// Displays everyone's stock
+function showAllStock(tableSvc, channel) {
+    const query = new azure.TableQuery().where('RowKey eq ?', '1');
+    let displayResult = 'Full ledger: ';
+
+    tableSvc.queryEntities(TABLENAME, query, null, function (error, result, response) {
+        if (!error) {
+            // query successful
+            for (let i = 0, len = result.entries.length; i < len; i++) {
+                let line = `${result.entries[i].PartitionKey._} has ${result.entries[i].schreefstock._}$$`;
+                displayResult = `${displayResult}\n${line}`;
+            }
+            displayResult = `${displayResult}\nschreef has an infinite amount of stock`;
+
+            channel.send(displayResult);
+        }
+        else {
+            LOG.error(error);
+        }
+    })
+};
+
+
+// Gets the amount of stock the user has and makes it available in the callback
+function getStock(tableSvc, username, callback) {
+    const query = new azure.TableQuery().select(['schreefstock']).where('PartitionKey eq ?', username);
+
+    tableSvc.queryEntities(TABLENAME, query, null, function (error, result, response) {
+        if (!error) {
+            // query successful
+            if (result.entries.length > 0) {
+                callback(result.entries[0].schreefstock._);
+            }
+            else {
+                callback(null);
```

Comment:
typically in a callback, in a failure, you pass both `null` and `error` so the callback can handle appropriately

Reply:
i'll add these